renderdoc/os/win32/win32_shellext.cpp

```diff
@@ -362,18 +399,219 @@ struct RDCThumbnailProvider : public IThumbnailProvider, IInitializeWithStream
+        case ResourceFormatType::R9G9B9E5:
+        case ResourceFormatType::Regular:
+        case ResourceFormatType::S8:
+        case ResourceFormatType::Undefined:
+        case ResourceFormatType::YUV10:
+        case ResourceFormatType::YUV12:
+        case ResourceFormatType::YUV16:
+        case ResourceFormatType::YUV8: return E_NOTIMPL;
+      }
+      thumbwidth = m_ddsData.width;
+      thumbheight = m_ddsData.height;
+      // Delete buffers
+      delete[] m_ddsData.subdata;
+      delete[] m_ddsData.subsizes;
+
+      byte *decompressed =
+          new byte[thumbheight * thumbwidth * 4];    // Decompressed DDS, 4 byte/pixel
+
+      unsigned char decompressedBlock[decompressedBlockMaxSize];
+      unsigned char greenBlock[16];
+      auto compBlockStart = m_Thumb.pixels;
+      for(uint32_t blockY = 0; blockY < thumbheight / 4; blockY++)
+      {
+        for(uint32_t blockX = 0; blockX < thumbwidth / 4; blockX++)
+        {
+          uint32_t decompressedBlockStart =
+              blockY * 4 * decompressedStride + blockX * decompressedBlockWidth;
+          switch(ddsFormatType)
+          {
+            case ResourceFormatType::BC1:
+              DecompressBlockBC1(compBlockStart, decompressedBlock, nullptr);
+              break;
+            case ResourceFormatType::BC2:
+              DecompressBlockBC2(compBlockStart, decompressedBlock, nullptr);
+              break;
+            case ResourceFormatType::BC3:
+              DecompressBlockBC3(compBlockStart, decompressedBlock, nullptr);
+              break;
+            case ResourceFormatType::BC4:
+              DecompressBlockBC4(compBlockStart, decompressedBlock, nullptr);
+              break;
+            case ResourceFormatType::BC5:
+              DecompressBlockBC5(compBlockStart, decompressedBlock, greenBlock, nullptr);
+              break;
+            case ResourceFormatType::BC6:    // other formats
+            case ResourceFormatType::BC7:
+            case ResourceFormatType::A8:
+            case ResourceFormatType::ASTC:
+            case ResourceFormatType::D16S8:
+            case ResourceFormatType::D24S8:
+            case ResourceFormatType::D32S8:
+            case ResourceFormatType::EAC:
+            case ResourceFormatType::ETC2:
+            case ResourceFormatType::PVRTC:
+            case ResourceFormatType::R10G10B10A2:
+            case ResourceFormatType::R11G11B10:
+            case ResourceFormatType::R4G4:
+            case ResourceFormatType::R4G4B4A4:
+            case ResourceFormatType::R5G5B5A1:
+            case ResourceFormatType::R5G6B5:
+            case ResourceFormatType::R9G9B9E5:
+            case ResourceFormatType::Regular:
+            case ResourceFormatType::S8:
+            case ResourceFormatType::Undefined:
+            case ResourceFormatType::YUV10:
+            case ResourceFormatType::YUV12:
+            case ResourceFormatType::YUV16:
+            case ResourceFormatType::YUV8: return E_NOTIMPL;
```

Comment:
Using `default:` here is fine for all the other formats I think, it keeps the code a bit more compact and easier to read.

Reply:
Changed now for all `switch(ResourceFormatType)` statements.